lib/ansible/modules/windows/win_dhcp_lease.ps1

```diff
@@ -0,0 +1,473 @@
+        $Object
+    )
+
+    $data = @{
+        address_state = $Object.AddressState
+        client_id     = $Object.ClientId
+        ip_address    = $Object.IPAddress.IPAddressToString
+        scope_id      = $Object.ScopeId.IPAddressToString
+        name          = $Object.Name
+        description   = $Object.Description
+    }
+
+    return $data
+}
+
+# Parse Regtype
+if ($dns_regtype) {
+    Switch ($dns_regtype) {
+        "aptr" { $dns_regtype = "AandPTR"; break }
+        "a" { $dns_regtype = "A"; break }
+        "noreg" { $dns_regtype = "NoRegistration"; break }
+        default { $dns_regtype = "NoRegistration"; break }
+    }
+}
+
+Try {
+    # Import DHCP Server PS Module
+    Import-Module DhcpServer
+}
+Catch {
+    # Couldn't load the DhcpServer Module
+    $module.FailJson("The DhcpServer module failed to load properly")
+}
+
+# Determine if there is an existing lease
+if ($ip) {
+    $current_lease = Get-DhcpServerv4Scope | Get-DhcpServerv4Lease | Where-Object IPAddress -eq $ip
+}
+
+# MacAddress was specified
+if ($mac) {
+    if ($mac -like "*-*") {
+        $mac_original = $mac
+        $mac = Convert-MacAddress -mac $mac
+    }
+
+    if ($mac -eq $false) {
+        $module.FailJson("The MAC Address is not properly formatted")
+    }
+    else {
+        $current_lease = Get-DhcpServerv4Scope | Get-DhcpServerv4Lease | Where-Object ClientId -eq $mac_original
+    }
+}
+
+# Did we find a lease/reservation
+if ($current_lease) {
+    $current_lease_exists = $true
+    $original_lease = $current_lease
+    $module.Diff.before = Convert-ReturnValue -Object $original_lease
+}
+else {
+    $current_lease_exists = $false
+}
+
+# If we found a lease, is it a reservation?
+if ($current_lease_exists -eq $true -and ($current_lease.AddressState -like "*Reservation*")) {
+    $current_lease_reservation = $true
+}
+else {
+    $current_lease_reservation = $false
+}
+
+# State: Absent
+# Ensure the DHCP Lease/Reservation is not present
+if ($state -eq "absent") {
+
+    # Required: MAC or IP address
+    if ((-not $mac) -and (-not $ip)) {
+        $module.Result.changed = $false
+        $module.FailJson("The ip or mac parameter is required for state=absent")
+    }
+
+    # If the lease exists, we need to destroy it
+    if ($current_lease_reservation -eq $true) {
+
+        # Try to remove reservation
+        Try {
+            $current_lease | Remove-DhcpServerv4Reservation -WhatIf:$check_mode
+            $state_absent_removed = $true
+        }
+        Catch {
+            $state_absent_removed = $false
+        }
+    }
+    else {
+
+        # Try to remove lease
+        Try {
+            $current_lease | Remove-DhcpServerv4Lease -WhatIf:$check_mode
+            $state_absent_removed = $true
+        }
+        Catch {
+            $state_absent_removed = $false
+        }
+    }
+
+    # If the lease doesn't exist, our work here is done
+    if ($current_lease_exists -eq $false) {
+        $module.Result.changed = $false
```

Comment:
No need to set, only do `$module.Result.changed = $true` if a change has been made.